core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup_test.go

```diff
@@ -0,0 +1,756 @@
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/feed_lookup_compatible_interface"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/i_keeper_registry_master_wrapper_2_1"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/models"
+)
+
+// setups up an evm registry for tests.
+func setupEVMRegistry(t *testing.T) *EvmRegistry {
+	lggr := logger.TestLogger(t)
+	addr := common.HexToAddress("0x6cA639822c6C241Fa9A7A6b5032F6F7F1C513CAD")
+	keeperRegistryABI, err := abi.JSON(strings.NewReader(i_keeper_registry_master_wrapper_2_1.IKeeperRegistryMasterABI))
+	require.Nil(t, err, "need registry abi")
+	feedLookupCompatibleABI, err := abi.JSON(strings.NewReader(feed_lookup_compatible_interface.FeedLookupCompatibleInterfaceABI))
+	require.Nil(t, err, "need mercury abi")
+	var headTracker httypes.HeadTracker
+	var headBroadcaster httypes.HeadBroadcaster
+	var logPoller logpoller.LogPoller
+	mockRegistry := mocks.NewRegistry(t)
+	mockHttpClient := mocks.NewHttpClient(t)
+	client := evmClientMocks.NewClient(t)
+
+	r := &EvmRegistry{
+		HeadProvider: HeadProvider{
+			ht:     headTracker,
+			hb:     headBroadcaster,
+			chHead: make(chan ocr2keepers.BlockKey, 1),
+		},
+		lggr:     lggr,
+		poller:   logPoller,
+		addr:     addr,
+		client:   client,
+		txHashes: make(map[string]bool),
+		registry: mockRegistry,
+		abi:      keeperRegistryABI,
+		active:   make(map[string]activeUpkeep),
+		packer:   &evmRegistryPackerV2_1{abi: keeperRegistryABI},
+		headFunc: func(ocr2keepers.BlockKey) {},
+		chLog:    make(chan logpoller.Log, 1000),
+		mercury: &MercuryConfig{
+			cred: &models.MercuryCredentials{
+				URL:      "https://google.com",
+				Username: "FakeClientID",
+				Password: "FakeClientKey",
+			},
+			abi:            feedLookupCompatibleABI,
+			allowListCache: cache.New(DefaultAllowListExpiration, CleanupInterval),
+		},
+		hc: mockHttpClient,
+	}
+	return r
+}
+
+// helper for mocking the http requests
+//func (r *EvmRegistry) buildRevertBytesHelper() []byte {
+//	mercuryErr := r.mercury.abi.Errors["FeedLookup"]
+//	mercuryLookupSelector := [4]byte{0x62, 0xe8, 0xa5, 0x0d}
+//	ml := FeedLookup{
+//		feedParamKey:  "feedIDHex",
+//		feeds:      []string{"0x4554482d5553442d415242495452554d2d544553544e45540000000000000000", "0x4254432d5553442d415242495452554d2d544553544e45540000000000000000"},
+//		timeParamKey: "blockNumber",
+//		time:      big.NewInt(8586948),
+//		extraData:  []byte{},
+//	}
+//	// check if Pack does not add selector for me
+//	pack, err := mercuryErr.Inputs.Pack(ml.feedParamKey, ml.feeds, ml.timeParamKey, ml.time, ml.extraData)
+//	if err != nil {
+//		log.Fatal("failed to build revert")
+//	}
+//	var payload []byte
+//	payload = append(payload, mercuryLookupSelector[:]...)
+//	payload = append(payload, pack...)
+//	return payload
+//}
+
+func TestEvmRegistry_mercuryLookup(t *testing.T) {
```

Comment:
remove tests if we arent using but have at least happy path shown

Reply:
done